Use with `postinstall`

I'm trying to use this package to install `peerDependencies` for [my shared config](https://github.com/okonet/eslint-config-okonet) in a config consumer repository using the `postinstall` script like this: https://github.com/okonet/eslint-config-okonet/pull/4/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R11 

but this produces the following error:

```
➜ npmD okonet/eslint-config-okonet#install-deps

> eslint-config-okonet@0.0.0-development postinstall /Users/okonet/Projects/OSS/attr-accept/node_modules/eslint-config-okonet
> install-peerdeps eslint-config-okonet --dev

install-peerdeps v1.1.3
Installing peerdeps for eslint-config-okonet@6.0.1.
npm install eslint-config-okonet eslint@4.x prettier@* --save-dev

npm ERR! code ENOSELF
npm ERR! Refusing to install package with name "eslint-config-okonet" under a package
npm ERR! also called "eslint-config-okonet". Did you name your project the same
npm ERR! as the dependency you're installing?
npm ERR! 
npm ERR! For more information, see:
npm ERR!     <https://docs.npmjs.com/cli/install#limitations-of-npms-install-algorithm>

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/okonet/.npm/_logs/2017-09-12T17_27_11_114Z-debug.log
ERR The install process exited with error code 1.
npm WARN eslint-config-okonet@0.0.0-development requires a peer of babel-eslint@^7.2.3 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-okonet@0.0.0-development requires a peer of eslint-config-airbnb@^15.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-okonet@0.0.0-development requires a peer of eslint-config-prettier@^2.3.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-okonet@0.0.0-development requires a peer of eslint-plugin-flowtype@^2.19.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-okonet@0.0.0-development requires a peer of eslint-plugin-import@^2.7.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-okonet@0.0.0-development requires a peer of eslint-plugin-jsx-a11y@^5.1.1 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-okonet@0.0.0-development requires a peer of eslint-plugin-node@^5.1.1 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-okonet@0.0.0-development requires a peer of eslint-plugin-prettier@^2.0.1 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-okonet@0.0.0-development requires a peer of eslint-plugin-react@^7.3.0 but none is installed. You must install peer dependencies yourself.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! eslint-config-okonet@0.0.0-development postinstall: `install-peerdeps eslint-config-okonet --dev`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the eslint-config-okonet@0.0.0-development postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/okonet/.npm/_logs/2017-09-12T17_27_11_762Z-debug.log
```

I'm not sure why it's trying to install the package itself and if this is by design.

--- a/src/cli.js
+++ b/src/cli.js
@@ -154,7 +154,7 @@ function installCb(err) {
   if (program.onlyPeers) {
     successMessage = `${successText} The peerDeps of ${packageName} were installed successfully.`;
   }
-  console.log();
+  console.log(successMessage);
   process.exit(0);
 }
 /* eslint-enable */

--- a/src/cli.js
+++ b/src/cli.js
@@ -32,6 +32,7 @@ program
   .version(version)
   .description('Installs the specified package along with correct peerDeps.')
   .option('-d, --dev', 'Install the package as a devDependency')
+  .option('-o, --onlyPeers', 'Install only peerDpenendncies of the package')
   .option('-S, --silent', 'If using npm, don\'t save in package.json')
   .option('-Y, --yarn', 'Install with yarn')
   .usage('<package>[@<version>], default version is \'latest\'')
@@ -108,6 +109,7 @@ const options = {
   packageName,
   version: packageVersion || 'latest',
   dev: program.dev,
+  onlyPeers: program['only-peers'],
   silent: program.silent,
   packageManager,
 };

--- a/src/cli.js
+++ b/src/cli.js
@@ -32,7 +32,7 @@ program
   .version(version)
   .description('Installs the specified package along with correct peerDeps.')
   .option('-d, --dev', 'Install the package as a devDependency')
-  .option('-o, --onlyPeers', 'Install only peerDpenendncies of the package')
+  .option('-o, --only-peers', 'Install only peerDependencies of the package')
   .option('-S, --silent', 'If using npm, don\'t save in package.json')
   .option('-Y, --yarn', 'Install with yarn')
   .usage('<package>[@<version>], default version is \'latest\'')
@@ -109,7 +109,7 @@ const options = {
   packageName,
   version: packageVersion || 'latest',
   dev: program.dev,
-  onlyPeers: program['only-peers'],
+  onlyPeers: program.onlyPeers,
   silent: program.silent,
   packageManager,
 };

--- a/src/cli.js
+++ b/src/cli.js
@@ -149,7 +149,12 @@ function installCb(err) {
     console.log(`${errorText} ${err.message}`);
     process.exit(1);
   }
-  console.log(`${successText} ${packageName} and its peerDeps were installed successfully.`);
+  let successMessage =
+    `${successText} ${packageName} and its peerDeps were installed successfully.`;
+  if (program.onlyPeers) {
+    successMessage = `${successText} The peerDeps of ${packageName} were installed successfully.`;
+  }
+  console.log();
   process.exit(0);
 }
 /* eslint-enable */

--- a/src/installPeerDeps.js
+++ b/src/installPeerDeps.js
@@ -61,7 +61,7 @@ function spawnInstall(command, args) {
   });
 }
 
-function installPeerDeps({ packageName, version, packageManager, dev, silent }, cb) {
+function installPeerDeps({ packageName, version, packageManager, dev, onlyPeers, silent }, cb) {
   // Thanks https://github.com/unpkg/npm-http-server/blob/master/modules/RegistryUtils.js
   // for scoped modules help
   let encodedPackageName;
@@ -96,7 +96,7 @@ function installPeerDeps({ packageName, version, packageManager, dev, silent }, 
       }
 
       // Construct packages string with correct versions for install
-      let packagesString = `${packageName}`;
+      let packagesString = onlyPeers ? '' : `${packageName}`;
       Object.keys(peerDepsVersionMap).forEach((depName) => {
         const range = peerDepsVersionMap[depName];
         // Semver ranges can have a join of comparator sets

--- a/src/installPeerDeps.js
+++ b/src/installPeerDeps.js
@@ -141,6 +141,12 @@ function installPeerDeps({ packageName, version, packageManager, dev, onlyPeers,
         args = args.concat('--save');
       }
 
+      // Remove empty args
+      // There's a bug with Yarn 1.0 in which an empty arg
+      // causes the install process to fail with a "malformed
+      // response from the registry" error
+      args = args.filter(a => a !== '');
+
       //  Show user the command that's running
       console.log(`Installing peerdeps for ${packageName}@${version}.`);
       console.log(`${packageManager} ${args.join(' ')}\n`);

--- a/src/installPeerDeps.js
+++ b/src/installPeerDeps.js
@@ -96,6 +96,7 @@ function installPeerDeps({ packageName, version, packageManager, dev, onlyPeers,
       }
 
       // Construct packages string with correct versions for install
+      // If onlyPeers option is true, don't install the package itself, only its peers.
       let packagesString = onlyPeers ? '' : `${packageName}`;
       Object.keys(peerDepsVersionMap).forEach((depName) => {
         const range = peerDepsVersionMap[depName];

